Description: kernel versionning enhancement
 Fix kernel version detection so that it is more flexible and
 less error prone.
 Thanks to Cristian Ionescu-Idbohrn <cristian.ionescu-idbohrn@axis.com>
 for reporting the issue and suggesting a patch.
 Patch slightly revisited by Jerome Benoit <calculus@rezozer.net>.
Origin: debian
Author: Cristian Ionescu-Idbohrn <cristian.ionescu-idbohrn@axis.com>
Last-Update: 2013-05-13

--- a/firehol.sh
+++ b/firehol.sh
@@ -5773,8 +5773,16 @@
 	exit 0
 fi
 
-KERNELMAJ=`${UNAME_CMD} -r | ${SED_CMD}                   -e 's,\..*,,'`
-KERNELMIN=`${UNAME_CMD} -r | ${SED_CMD} -e 's,[^\.]*\.,,' -e 's,\..*,,'`
+kernel_maj_min() {
+	local kmaj kmin IFS=.-
+
+	kmaj=$1
+	kmin=$2
+
+	set -- $(${UNAME_CMD} -r)
+	eval $kmaj=\$1 $kmin=\$2
+}
+kernel_maj_min KERNELMAJ KERNELMIN
 
 if [ "$KERNELMAJ" -lt 2 ] ; then
 	echo >&2 "FireHOL requires a kernel version higher than 2.3."
